packages/trader/src/Modules/Trading/Components/Form/Purchase/contract-info.tsx

```diff
@@ -1,44 +1,26 @@
 import classNames from 'classnames';
-import PropTypes from 'prop-types';
 import React from 'react';
-import { DesktopWrapper, Icon, MobileWrapper, Money, Popover, Text } from '@deriv/components';
+import { DesktopWrapper, MobileWrapper, Money, Popover, Text } from '@deriv/components';
 import { Localize, localize } from '@deriv/translations';
 import { getContractSubtype, getCurrencyDisplayCode, getLocalizedBasis, getGrowthRatePercentage } from '@deriv/shared';
-import CancelDealInfo from './cancel-deal-info.jsx';
+import CancelDealInfo from './cancel-deal-info';
+import ValueMovement from './value-movement';
+import { TProposalTypeInfo } from 'Types';
 
-export const ValueMovement = ({
-    has_error_or_not_loaded,
-    proposal_info,
-    currency,
-    has_increased,
-    is_turbos,
-    is_vanilla,
-    value,
-    show_currency = true,
-}) => (
-    <div className='price-info--value-container'>
-        <div className='trade-container__price-info-value'>
-            {!has_error_or_not_loaded && (
-                <Money
-                    amount={proposal_info?.obj_contract_basis?.value || value}
-                    className={classNames('trade-container__price-info-currency', {
-                        'trade-container__price-info-currency--payout-per-point': is_vanilla || is_turbos,
-                    })}
-                    currency={currency}
-                    should_format={!is_turbos && !is_vanilla}
-                    show_currency={show_currency}
-                />
-            )}
-        </div>
-        <div className='trade-container__price-info-movement'>
-            {!has_error_or_not_loaded && has_increased !== null && has_increased ? (
-                <Icon icon='IcProfit' />
-            ) : (
-                <Icon icon='IcLoss' />
-            )}
-        </div>
-    </div>
-);
+type TContractInfo = {
+    basis: string;
+    currency: string;
+    growth_rate: number;
+    has_increased: boolean;
+    is_accumulator: boolean;
+    is_multiplier: boolean;
+    is_turbos: boolean;
```

Comment:
and here extract prop types from `ValueMovement` component instead of manually typing.

`Pick<React.ComponentProps<typeof ValueMovement>, 'is_turbos' | 'is_vanilla' | '......etc'>`

If the same prop goes into multiple components then you can do

`Pick<React.ComponentProps<typeof ValueMovement>, 'is_turbos' | 'is_vanilla' | '......etc'>` | `Pick<React.ComponentProps<typeof SomeComponent>, 'is_turbos' | 'is_vanilla' | '......etc'>`

It'll ensure that types comes from single source and typescript will complain accordingly later on.